### DISCUSSION TITLE
Multiple batteries not shown properly?

### DISCUSSION THREAD

**Discussion by blauwkeut** [JanDeDobbeleer/oh-my-posh#620] [Troubleshoot]
Hi Jan,

My laptop has two batteries, but oh-my-posh seems to show this in a wierd way. Not sure if it's oh-my-posh itself or the battery library, but the prompt only shows the status of one of the batteries...

![2021-04-07 14_12_10-Administrator_ Windows PowerShell](https://user-images.githubusercontent.com/33721771/113864717-89152c00-97ab-11eb-9863-75010f4fe51f.png)

![2021-04-07 14_13_28-Administrator_ Windows PowerShell](https://user-images.githubusercontent.com/33721771/113864724-89adc280-97ab-11eb-81ac-265f6c297138.png)

Is there a way to show the status of each battery, or the 'sum' of both batteries as reported by the OS? Or both so we can choose...
Tried messing with the template and trying to reference it array-like .Battery[1] but that didn't work.

Second question: what are the tresholds for color changes on this? Certain percentages of remaining charge? Can this be modified (colors per percentage)? And could I e.g. show a different icon based on remaining capacity (using the different battery level icons in NerdFont)?

Cheers!

**Comment by JanDeDobbeleer** [comment]
You're right, we need to count them all. Don't do that yet. I'll fix that over the weekend. When it comes to the colors you can make use of the [template](https://ohmyposh.dev/docs/battery#template-properties) functionality in the [colors](https://ohmyposh.dev/docs/configure#foreground-templates).

**Reply by blauwkeut** [reply]
Cool, thanks for looking into that...

Also messing about then with the template functionality (spending way too much time on this probably, since I have no experience with Go templating) and getting somewhere:

![2021-04-08 10_15_07-C__scripts](https://user-images.githubusercontent.com/33721771/113992512-d5b44200-9853-11eb-8095-27480a6cc296.png)

Does templating work everywhere? Tried adding the same template to "background"/"foreground" to have the diamond bits also change color, but that seems to be ignored?

When I tried to change the color for Charging through the template, I get an error. Maybe because I don't understand the template functionality but I seem to have it like I have found in some tutorials:

I've got an if like this:
`{{ if eq .Battery.State \"Charging\" }}`
or
`{{ if contains .Battery.State \"Charging\" }}`

Which results in this:
![2021-04-08 10_16_23-C__scripts](https://user-images.githubusercontent.com/33721771/113992921-3479bb80-9854-11eb-85bb-c8aaa7471f40.png)

Any thoughts?

Thanks,
Dennis

**Reply by JanDeDobbeleer** [reply]
Allow me to have a look tomorrow, it should work (although I have to admit I didn't check the diamond type when adding the color templates 😱).

**Reply by blauwkeut** [reply]
👍 no worries - not business critical at all so whenever you've got time...

**Reply by JanDeDobbeleer** [reply]
@blauwkeut I just had a look (batteries will be counted so that should be available soon) and the following does what you want:

```json
{
  "type": "battery",
  "style": "diamond",
  "leading_diamond": "\ue0b6",
  "trailing_diamond": "\ue0b4",
  "foreground": "#ffffff",
  "background": "#CCFF00",
  "background_templates": [
    "{{if ge .Percentage 90}}#990066{{else if ge .Percentage 70}}#009933{{end}}"
  ],
  "properties": {
    "battery_icon": "",
    "discharging_icon": " ",
    "charging_icon": " ",
    "charged_icon": " ",
    "postfix": " "
  }
}
```

**Reply by blauwkeut** [reply]
Tried background_templates and that indeed does the trick...

One last question then maybe if you don't mind: with the template I can't seem to get a 'state' for charging. I have the following (shortened version for clarity):
`"template": "{{ $B := .Percentage }}{{ if eq .Battery.State \"Charging\" }}<#ffffff,#40c4ff>\uf1e6{{ else if le $B 5 }}<#ffffff,#DA2C00>\uf582{{end}} {{$B}}\uF295</>"`

But this gives an error - presumably the escaped quotes around 'Charging' are causing this?:
![2021-04-15 08_49_48-~](https://user-images.githubusercontent.com/33721771/114826289-9a6fc100-9dc7-11eb-8f3b-413bfb2e8c68.png)
(also tried contains instead of eq, but same result)

If I just display .Battery.State to the prompt I do get 'Charging' so it seems to return the right value, but the comparison doesn't work?

**Reply by JanDeDobbeleer** [reply]
The actual error is the following:

`s:"template: title:1:29: executing "title" at <eq .Battery.State "Charging">: error calling eq: incompatible types for comparison"`

That implies that a string value can't be compared against an int value (state is actually an int). Change it to this and it will work:

`"template": "{{ $B := .Percentage }}{{ if eq .Battery.State.String \"Charging\" }}<#ffffff,#40c4ff>\uf1e6{{ else if le $B 5 }}<#ffffff,#DA2C00>\uf582{{end}} {{$B}}\uF295</>"`

**Reply by blauwkeut** [reply]
A simple cast... I didn't see that mentioned anywhere on the Go templating docs I read, but there you go...

![2021-04-16 16_28_36-C__scripts](https://user-images.githubusercontent.com/33721771/115039858-1f92cd00-9ed1-11eb-9d20-879bb2ba1c6b.png)

For completeness sake, and for anybody else wanting this or similar, below is the theme code - adapt to suit your needs :smile: - this is basically from the jandedobbeleer.omp.json, but add the "background_templates" and "template" under "properties".

        {
          "type": "battery",
          "style": "powerline",
          "invert_powerline": true,
          "powerline_symbol": "\uE0B2",
          "foreground": "#ffffff",
          "background": "#f36943",
          "background_templates": [
             "{{ $B := .Percentage }}{{ if eq .Battery.State.String \"Full\" }}#00c94a{{ else if eq .Battery.State.String \"Charging\" }}#40c4ff{{ else if le $B 5 }}#DA2C00{{ else if le $B 20 }}#FF4E22{{ else if le $B 40 }}#FF953F{{ else if le $B 60 }}#FFC73A{{ else if le $B 80 }}#00B9A3{{ else if le $B 100 }}#005F84{{end}}"
          ],
          "properties": {
            "template": "{{ $B := .Percentage }}{{ if eq .Battery.State.String \"Full\" }}<#193549,#00c94a>\uf1e6{{ else if eq .Battery.State.String \"Charging\" }}<#193549,#40c4ff>\uf583{{ else if le $B 5 }}<#ffffff,#DA2C00>\uf582{{ else if le $B 20 }}<#193549,#FF4E22>\uf579{{ else if le $B 40 }}<#193549,#FF953F>\uf57b{{ else if le $B 60 }}<#193549,#FFC73A>\uf57d{{ else if le $B 80 }}<#193549,#00B9A3>\uf57f{{ else if le $B 100 }}<#ffffff,#005F84>\uf578{{end}} {{$B}}\uF295</>",
            "postfix": " "
          }
        },
